lib/error.js

```diff
@@ -0,0 +1,42 @@
+/** takes an error object and serializes it to a plan object */
+function serializedError (input) {
+  if (!(input instanceof Error)) {
+    const error = new Error(`attempted to serialize a non-error, ${typeof input} ${input?.constructor?.name}`)
+    return serializedError(error)
+  }
+  // CAUTION: what we serialize here gets add to the size of logs
+  return {
+    errorType: input.errorType ?? input.constructor?.name,
+    ...(input.message ? { message: input.message } : {}),
```

Comment:
`JSON.stringify` and `bole` v4+ drop undefined values. See below.

But if you feel you need to filter out undefined values: What if `input.constructor` is undefined?

```
> JSON.stringify({ a: 42, b: undefined })
'{"a":42}'
```

Reply:
`input.constructor` cant be undefined because at this point it's a guarded error object. input will always be of type error, `errorType` may be undefined, but the constructor will always be there with a name. I can remove the `?` in `constructor?` but I cant make `errorType` conditional like the others because there's no way to create an error without a constructor or without a name so I couldn't cover it, I did make a concerted effort before seeing your post, and gave up trying to handle edge cases that may never exist. we can do `?? "Error"` at the end, and add in a coverage ignore comment.

Reply:
If you're confident `input.constructor.name` will always be defined, simply removing the `?` in `constructor?` sounds good to me!

Reply:
done